src/db/collections/outputs/mod.rs

```diff
@@ -213,6 +214,68 @@ impl OutputCollection {
         Ok(())
     }
 
+    /// Create the initial ledger updates materialized view by pulling all output data.
+    #[instrument(skip_all, err, level = "trace")]
+    pub async fn create_ledger_updates(&self) -> Result<(), Error> {
+        self.aggregate::<OutputDocument>(
+            vec![
+                doc! { "$match": { "details.address": { "$ne": null } } },
+                doc! { "$project": {
+                    "_id": {
+                        "milestone_index": { "$ifNull": [
+                            "$metadata.spent_metadata.spent.milestone_index",
+                            "$metadata.booked.milestone_index"
+                        ] },
+                        "output_id": "$_id",
+                        "is_spent": { "$ne": [ "$metadata.spent_metadata", null ] },
+                    },
+                    "address": "$details.address",
+                    "milestone_timestamp": "$metadata.booked.milestone_timestamp",
+                } },
+                doc! { "$merge": { "into": LedgerUpdateCollection::NAME, "whenMatched": "keepExisting" } },
+            ],
+            None,
+        )
+        .await?
+        .try_next()
+        .await?;
+        Ok(())
+    }
+
+    /// Merges the outputs table into the ledger updates materialized view.
+    #[instrument(skip_all, err, level = "trace")]
+    pub async fn merge_into_ledger_updates(&self, milestone_index: MilestoneIndex) -> Result<(), Error> {
+        self.aggregate::<OutputDocument>(
+            vec![
+                doc! { "$match": {
+                    "details.address": { "$ne": null },
+                    "$or": [
+                        { "metadata.booked.milestone_index": milestone_index },
+                        { "metadata.spent_metadata.spent.milestone_index": milestone_index },
+                    ]
+                } },
+                doc! { "$project": {
+                    "_id": {
+                        "milestone_index": { "$ifNull": [
+                            "$metadata.spent_metadata.spent.milestone_index",
+                            "$metadata.booked.milestone_index"
+                        ] },
+                        "output_id": "$_id",
+                        "is_spent": { "$ne": [ "$metadata.spent_metadata", null ] },
+                    },
+                    "address": "$details.address",
+                    "milestone_timestamp": "$metadata.booked.milestone_timestamp",
```

Comment:
Does that always have to be the `timestamp` of the "booking" milestone?

Reply:
Ah, good catch.

Reply:
Good catch, I think you're right. This should probably use the `$ifNull` mechanism too.

@Alexandcoats Could you please at this to your test case as well?